### PR TITLE
Feature/edit delete flight

### PR DESCRIPTION
**Pull request by NicoleHoerner**

### CODE REVIEW

File: components/Form/AddFlightForm.js [hunk 1]
```diff
@@ -21,12 +21,12 @@ export default function AddFlightForm() {
 
   const router = useRouter();
 
-  const [isFocused, setIsFocused] = useState({});
+  const [$isFocused, set$isFocused] = useState({});
```
Comment (Kheirah) [inline components/Form/AddFlightForm.js:24]: The state variable doesn't need the dollar sign. Only the prop on the styled components needs it.

File: components/Form/AddFlightForm.js [hunk 2]
```diff
@@ -21,12 +21,12 @@ export default function AddFlightForm() {
...
 
-  const handleFocus = (event) => setIsFocused({ [event.target.name]: true });
+  const handleFocus = (event) => set$isFocused({ [event.target.name]: true });
   const handleBlur = (event) => {
     console.log("Blur event fired:", event.target.value);
```
Comment (Kheirah) [inline components/Form/AddFlightForm.js:28]: Remove `console.logs` when you're done inspecting the behavior of your app,

File: components/Form/AddFlightForm.js [hunk 3]
```diff
@@ -147,10 +150,10 @@ export default function AddFlightForm() {
 
   const router = useRouter();
```
Comment (Kheirah) [inline components/Form/AddFlightForm.js:151]: Delete the entire block of code that is commented out. Don't keep duplicate code around.

File: components/Modal/EditFlightModal.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    onSave(editedFlightData);
+    onClose();
+  };
+
+  const modalContent = (
+    <ModalOverlay onClose={onClose}>
+      <ModalContainer>
+        <form>
+          <ModalHeader>Edit Flight</ModalHeader>
```
Comment (Kheirah) [inline components/Modal/EditFlightModal.js:58]: The submit event is triggered by the submit button, but it's caught by the form itself.
```suggestion
        <form onSubmit={handleSaveChanges}>
```

File: components/Modal/EditFlightModal.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+              id="arrival"
+              name="arrival"
+              defaultValue={arrival}
+              onChange={handleArrivalChange}
+            />
+            {/* Add similar input fields for other flight details */}
+          </ModalBody>
+          <ModalFooter>
+            <SubmitButton type="submit" onClick={handleSaveChanges}>
+              Save Changes
```
Comment (Kheirah) [inline components/Modal/EditFlightModal.js:93]: The function of the `onClick` event goes up to the `onSubmit` event of the form above. 
```suggestion
            <SubmitButton type="submit">
```

File: components/Modal/EditFlightModal.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    const editedFlightData = {
+      flight_iata: flightIata,
+      scheduled_date: scheduledDate,
+      departure: departure,
+      arrival: arrival,
+    };
+
+    onSave(editedFlightData);
+    onClose();
+  };
```
Comment (Kheirah) [inline components/Modal/EditFlightModal.js:52]: The event parameter was missing. What is more, I added additional info about the original iata and date of the flight to uniquely identify the entry in local storage. You can now change the iata and date as well.
```suggestion
  const handleSaveChanges = (event) => {
    event.preventDefault();
    const editedFlightData = {
      flight_iata: flightIata,
      scheduled_date: scheduledDate,
      departure: departure,
      arrival: arrival,
    };

    onSave(selectedFlight.flight_iata, selectedFlight.scheduled_date, editedFlightData);
    onClose();
  };
```

File: pages/flight_list.js [hunk 7]
```diff
@@ -12,9 +17,69 @@ export default function FlightList() {
...
+    setSelectedFlight(selectedFlight);
+  };
+
+  const handleEditModalSave = (editedFlightData) => {
+    // Update the flight data in main state (flightsOfInterest)
+    const updatedFlights = flightsOfInterest.map((flight) => {
+      if (
+        flight.flight_iata === editedFlightData.flight_iata &&
+        flight.scheduled_date === editedFlightData.scheduled_date
+      ) {
```
Comment (Kheirah) [inline pages/flight_list.js:51]: The additional info is accepted here and passed along to check for the correct flight entry.
```suggestion
  const handleEditModalSave = (iata, date, editedFlightData) => {
    // Update the flight data in main state (flightsOfInterest)
    const updatedFlights = flightsOfInterest.map((flight) => {
      if (
        flight.flight_iata === iata &&
        flight.scheduled_date === date
```

**Review by Kheirah — CHANGES_REQUESTED** [review]

File: components/Modal/EditFlightModal.js (hunk 4 above)
Comment (NicoleHoerner) [reply components/Modal/EditFlightModal.js:58]: I inverted the lines and added `onSubmit={handleSaveChanges}` to the form:
```
<ModalHeader>Edit Flight</ModalHeader>
<form onSubmit={handleSaveChanges}>
```

**Review by Kheirah — APPROVED** [review]
LGTM!

**Pull request merged**